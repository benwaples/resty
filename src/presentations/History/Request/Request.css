.Request {
  display: inline;
  .method {
    background-color: #52A0A3;
    color: hsl(142, 100%, 98%);
    width: fit-content;
    padding: 5px;
    margin: 0;
    display: inline;
    border-radius: 5px 0 0 5px;
  }
  .url {
    background-color: #000F08;
    color: #F4FFF8;
    padding: 5px;
    margin: 0;
    display: inline;
    
  }
}