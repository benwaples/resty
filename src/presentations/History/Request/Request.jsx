import React from 'react';
import PropTypes from 'prop-types';
import styles from './Request.css';

function Request({ method, url }) {
  return (
    <li className={styles.Request}>
      <p>{method}</p>
      <p>{url}</p>
    </li>
  );
}

Request.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Request;

