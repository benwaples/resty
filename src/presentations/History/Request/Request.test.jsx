import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Request from './Request';

describe('Old request component', () => {
  afterEach(() => cleanup());

  it('should render an old request', () => {
    const { asFragment } = render(<Request 
      method="GET"
      url="google.com" 
      onClick={() => {}}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
