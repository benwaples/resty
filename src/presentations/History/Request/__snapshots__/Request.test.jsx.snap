// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Old request component should render an old request 1`] = `
<DocumentFragment>
  <li
    class="Request"
  >
    <p>
      GET
    </p>
    <p>
      google.com
    </p>
  </li>
</DocumentFragment>
`;
